docs/common/custom_column_types.md

```diff
@@ -31,6 +32,24 @@ val queryWrapper: Database = Database(
 )
 ```
 
+## Primitives
+
+A sibling module that adapts primitives for your convenience.
+
+```groovy
+dependencies {
+  implementation "com.squareup.sqldelight:primitive-adapters:{{ versions.sqldelight }}"
+}
+```
+
+The following adapters exist:
+
+- `BooleanColumnAdapter` — Retrieves `kotlin.Boolean` for an SQL type implicitly stored as `kotlin.Long`
+- `DoubleColumnAdapter` — Retrieves `kotlin.Double` for an SQL type implicitly stored as `kotlin.Long`
+- `FloatColumnAdapter` — Retrieves `kotlin.Float` for an SQL type implicitly stored as `kotlin.Long`
+- `IntColumnAdapter` — Retrieves `kotlin.Int` for an SQL type implicitly stored as `kotlin.Long`
+- `ShortColumnAdapter` — Retrieves `kotlin.Short` for an SQL type implicitly stored as `kotlin.Long`
+
```

Comment:
Idk if this should be promoted for use in dialects other than SQLite, as HSQL, MySQL, and PostgreSQL all have native SQL types that map to the above (e.g., in PostgreSQL, a user shouldn't do `INT AS kotlin.Short` and use `ShortColumnAdapter` as PostgreSQL has the type `SMALLINT` which natively decodes to `kotlin.Short`). Should this be moved to the SQLite specific documentation (even though it can technically be used by other dialects)?

Reply:
SQLite specific